index/scorch/optimize.go

```diff
@@ -262,6 +262,10 @@ OUTER:
 			bm.And(actualBM)
 		}
 
+		// On closure, do not recycle this optimized TermFieldReader -
+		// as the roaring.Bitmap created above needs to be released.
+		oTFR.doNotRecycle = true
```

Comment:
not sure whether we need this extra field as it is already using a specific fieldName/term for all the optimisations.. So actual recycle logic could be somewhere inside the recycleTermFieldReader.

Lesser surface area for changes and no need to set this extra flag..

Reply:
Hmm, we can avoid recycling only when a new instance of roaring.Bitmap is created. There's several cases above where we just end up reusing the bitmaps and not setting up new instances.

Reply:
In all such applicable cases, we would have created a new IndexSnapshotTermFieldReaders with that customised field/terms? So it can still handled in a single place.

Another thought is -  Do we really need to throw away the newly created IndexSnapshotTermFieldReaders ?
Can't we enable the reuse for them too? ie can't we alloc them or ask from the index snapshot the same way we do for other TFRs?

Reply:
I am taking back my thought on re usability/recycling here. Mostly this will only add more IS TFRs pointers for GC to track and hence pain rather a win.